src/pytest_param_scope/plugin.py

```diff
@@ -58,8 +71,18 @@ def param_scope(request):
     yield __data.setup_value
 
     if __data.ready_for_teardown:
+        teardown_gen = __data.teardown_gen
         teardown_func = __data.teardown_func
+
         __data = ParamScopeData()  # reset for next one
+
+        if teardown_gen:
+            try:
+                next(teardown_gen)
+            except StopIteration:
+                pass # this is expected
+
+        # should we disallow both a teardown from gen and a teardown?
         if teardown_func:
             teardown_func()
 
```

Comment:
Disallowing excess teardown could be done here as you've indicated, but also could be handled earlier (see comment above)

Reply:
yeah. I'll do it earlier